File: submissions/0014-longest-common-prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if(strs.length == 1) return strs[0];
    let smallest = 201;
    for(let i=0; i<strs.length; i++){
        smallest = Math.min(smallest,strs[i].length);
    }
    let max = -1;
    for(let i=1; i<=smallest; i++){
        let flag = true;
        for(let j=0; j<strs.length-1; j++){
            if(strs[j].substring(0,i) != strs[j+1].substring(0,i)){
                flag = false;
                break;
            }
        }
        if(flag) max = i;
    }
    if(smallest == 1){
        let flag = true;
        for(let j=0; j<strs.length-1; j++){
            if(strs[j].substring(0,1) != strs[j+1].substring(0,1)){
                flag = false;
                break;
            }
        }
        if(flag) max = 1;
    }
    if(max == -1) return "";
    return strs[0].substring(0,max);
};

File: submissions/0014-longest-common-prefix/solution.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String s = "";
        int min = 201;
        for(String curr : strs){
            if(curr.length() < min){
                s = curr;
                min = curr.length();
            }
        }
        int i=0;
        for(i=0; i<s.length(); i++){
            for(int j=0; j<strs.length; j++){
                if(s.charAt(i) != strs[j].charAt(i)) return s.substring(0, i);
            }
        }
        return s.substring(0, i);
    }
}
